### PR TITLE
Create next example

### LINKED ISSUE #202: Create next example

### PR DESCRIPTION
**Pull request by kennylavender**
closes #202 

Create an example for [next](https://nextjs.org/)

### CODE REVIEW

**Comment by kennylavender** [comment]
There seems to be an issue when you go to /recently-active-users client side, it doesn't render the 404 error, it renders the next unexpected error, this is because we are not passing the component a statusCode. Il need to look into how best to provide the statusCode or maybe we can just default to 404.

**Comment by kennylavender** [comment]
WIP, this needs more work, don't bother reviewing atm.

**Comment by kennylavender** [comment]
Ready for review. I think it can be improved in a few ways, but this is a good starting point.

File: examples/next/source/features/recently-active-users-page/menu-link-component.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+const MenuLink = () => (
+  <Feature
+    name={RECENTLY_ACTIVE_USERS_PAGE}
+    inactiveComponent={() => null}
+    activeComponent={
+      () => 
+        <li>
+          {/*
+            Eventually we may add a function that applies currently activated
```
Comment (ericelliott) [inline examples/next/source/features/recently-active-users-page/menu-link-component.js]: Don't we have access to initial features from any component under the features provider in the component tree?

**Comment by ericelliott** [comment]
It looks like there's a whole lot of ad-hoc plumbing in the example that should probably be part of the library.  Does every user need to jump through all these hoops and create a dozen different files just to conditionally display a feature page or 404 page?

If so, why?

**Comment by ericelliott** [comment]
I'd like to see a minimal example of this in action using about 1/10th as much code. ;)

File: examples/next/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Next Example
```
Comment (ericelliott) [inline examples/next/README.md]: I think this example needs much better documentation. What is the purpose of all these different files? Why did it take 20 different files to demonstrate this feature? Is this it reasonable for a user of this library to see all these files and absorb what each one is doing and figure out to implement the feature in their application within 20-30 minutes?

It should be.

Comment (kennylavender) [reply examples/next/README.md]: Next requires a bunch more work and customization due to having to use getInitialProps.

We could build a library to supply these next specific HOC's, next-react-feature-toggles. I guess they could come from this library too, but I was avoiding making this specific to next at the start.

File: examples/next/source/features/recently-active-users-page/menu-link-component.js (hunk 1 above)
Comment (kennylavender) [reply examples/next/source/features/recently-active-users-page/menu-link-component.js]: We do, I haven't yet created an easy to use function to apply them to a given path. #174

**Comment by kennylavender** [comment]
> I'd like to see a minimal example of this in action using about 1/10th as much code. ;)

I agree, that was my goal when I started, but I realized writing this out like a real app helped me understand the strengths and weaknesses of the current API. I was thinking code like this might make good code for integration tests, we could run integration tests against it with cypress.io. Then we could have simpler example code. What do you think?

**Comment by ericelliott** [comment]
Check out how [RITEway separately exports a React tool](https://github.com/ericelliott/riteway/blob/master/render-component.js) to help React users without requiring all users to install and use React.

As a user, I think I'd prefer it to be colocated with this same library, but perhaps exported separately.

**Comment by kennylavender** [comment]
Closing, will go in new smaller example direction after creating new next specific code.

**Pull request closed without merging**